async-nats/src/lib.rs

```diff
@@ -807,6 +807,19 @@ impl Client {
         }
     }
 
+    pub async fn request_timeout(
```

Comment:
Open to discuss but this seems more of a utility thing that you can do without us intervening (as an ext trait) in core types.

Reply:
Yes, agree.

This can be useful, but let's have some kind of ext pattern for that.

Reply:
got it